zipline/utils/calendars/trading_schedule.py

```diff
@@ -0,0 +1,379 @@
+
+    @abstractproperty
+    def first_execution_day(self):
+        """
+        The first possible day of trading in this TradingSchedule.
+        """
+        raise NotImplementedError()
+
+    @abstractproperty
+    def last_execution_day(self):
+        """
+        The last possible day of trading in this TradingSchedule.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def trading_sessions(self, start, end):
+        """
+        Calculates all of the trading sessions between the given
+        start and end.
+
+        Parameters
+        ----------
+        start : Timestamp
+        end : Timestamp
+
+        Returns
+        -------
+        DataFrame
+            A DataFrame, with a DatetimeIndex of trading dates, containing
+            columns of trading starts and ends in this TradingSchedule.
+        """
+        raise NotImplementedError()
+
+    @property
+    @remember_last
+    def all_execution_days(self):
+        return self.schedule.index
+
+    @property
+    @remember_last
+    def all_execution_minutes(self):
+        return all_scheduled_minutes(self.all_execution_days,
+                                     self.execution_minutes_for_days_in_range)
+
+    def trading_dates(self, start, end):
+        """
+        Calculates the dates of all of the trading sessions between the given
+        start and end.
+
+        Parameters
+        ----------
+        start : Timestamp
+        end : Timestamp
+
+        Returns
+        -------
+        DatetimeIndex
+            A DatetimeIndex containing the dates of the desired trading
+            sessions.
+        """
+        return self.trading_sessions(start, end).index
+
+    @abstractmethod
+    def data_availability_time(self, date):
+        """
+        Given a UTC-canonicalized date, returns a time by-which all data from
+        the previous date is available to the algorithm.
+
+        Parameters
+        ----------
+        date : Timestamp
+            The UTC-canonicalized calendar date whose data availability time
+            is needed.
+
+        Returns
+        -------
+        Timestamp or None
+            The data availability time on the given date, or None if there is
+            no data availability time for that date.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def start_and_end(self, date):
+        """
+        Given a UTC-canonicalized date, returns a tuple of timestamps of the
+        start and end of the algorithm trading session for that date.
+
+        Parameters
+        ----------
+        date : Timestamp
+            The UTC-canonicalized algorithm trading session date whose start
+            and end are needed.
+
+        Returns
+        -------
+        (Timestamp, Timestamp)
+            The start and end for the given date.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def is_executing_on_minute(self, dt):
```

Comment:
Seems like this should be called `should_execute_on_minute`, given that that's the phrasing used everywhere in the docs and description.